src/weights.jl

```diff
@@ -382,6 +382,18 @@ Compute the weighted sum of an array `v` with weights `w`, optionally over the d
 """
 wsum(v::AbstractArray, w::AbstractVector, dims::Colon=:) = transpose(w) * vec(v)
 
+# Optimized methods (to ensure we use BLAS when possible)
+for W in (AnalyticWeights, FrequencyWeights, ProbabilityWeights, Weights)
+    @eval begin
+        wsum(v::AbstractArray, w::$W, dims::Colon) = transpose(w.values) * vec(v)
```

Comment:
Did you compare it with
```suggestion
        wsum(v::AbstractArray, w::$W, dims::Colon) = dot(w.values, v)
```
? I'm not sure if that would be faster or slower.

Reply:
Long story. :-) https://github.com/JuliaStats/StatsBase.jl/pull/737

Reply:
Is this relevant here? The weights are real numbers and hence transpose and conjugate transpose are the same.

Reply:
Ah I didn't know that. One could still use `dot` for `AbstractArray{<:Number}` if it is more performant.

Reply:
IIRC I benchmarked it for https://github.com/JuliaLang/Statistics.jl/pull/94 and it wasn't faster, otherwise I would have kept a specialized method for reals. As I noted there, these are equivalent anyway in common cases:
> Use `adjoint(y) * x` instead of `dot(y, x)`, as the former falls back to the latter for arrays of `Number`s but to `sum(uu*vv for (uu, vv) in zip(u, v))` for other types.